src/applications/Machine.java

```diff
@@ -39,35 +39,35 @@ public int getNumTasks() {
         return numTasks;
     }
 
-    public void setNumTasks(int numTasks) {
```

Comment:
I like how you cleared out several methods here and generally improved encapsulation.